features/admin.server-configurations.v1/models/password-recovery.ts

```diff
@@ -0,0 +1,170 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { GovernanceConnectorInterface } from "./governance-connectors";
+
+/**
+ * Proptypes for the Password Recovery Form error messages.
+ */
+export interface PasswordRecoveryFormErrorValidationsInterface {
+    /**
+     * Recovery link expiry time field.
+     */
+    expiryTime: string;
+    /**
+     * Sms otp expiry time field
+     */
+    smsOtpExpiryTime: string;
+    /**
+     * SMS OTP code length field.
+     */
+    smsOtpLength: string;
+    /**
+     * max resend count field.
+     */
+    maxResendCount: string;
+    /**
+     * Max allowed failed attempts count field.
+     */
+    maxFailedAttemptCount: string;
+}
+
+/**
+ * Form initial values interface.
+ */
+export interface PasswordRecoveryFormValuesInterface {
+    /**
+     * Recovery link expiry time.
+     */
+    expiryTime: string;
+    /**
+     * Notify user on successful password recovery.
+     */
+    notifySuccess: boolean;
+    /**
+     * Whether email based recovery is enabled.
+     */
+    enableEmailBasedRecovery: boolean;
+    /**
+     * Whether SMS based recovery is enabled.
+     */
+    enableSMSBasedRecovery: boolean;
+    /**
+     * SMS OTP expiry time.
+     */
+    smsOtpExpiryTime: string;
+    /**
+     * Whether to use upper case letters in SMS OTP code.
+     */
+    passwordRecoveryOtpUseUppercase: boolean;
+    /**
+     * Whether to use lower case letters in SMS OTP code.
+     */
+    passwordRecoveryOtpUseLowercase: boolean;
+    /**
+     * Whether to use numeric characters in SMS OTP code.
+     */
+    passwordRecoveryOtpUseNumeric: boolean;
+    /**
+     * The length of the SMS OTP code.
+     */
+    smsOtpLength: string;
+    /**
+     * The maximum amount of times recovery code/link is resent.
+     */
+    maxResendCount: string;
+    /**
+     * The maximum allowed failed attempts for a recovery flow.
+     */
+    maxFailedAttemptCount: string;
+}
+
+/**
+ * Interface for Password Recovery Configuration Form props.
+ */
+export interface PasswordRecoveryConfigurationFormPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Connector's initial values.
+     */
+    initialValues: GovernanceConnectorInterface;
+    /**
+     * Callback for form submit.
+     * @param values - Resolved Form Values.
+     */
+    onSubmit: (values) => void;
+    /**
+     * Is readonly.
+     */
+    readOnly?: boolean;
+    /**
+     * Whether the connector is enabled using the toggle button.
+     */
+    isConnectorEnabled?: boolean;
+    /**
+     * Specifies if the form is submitting.
+     */
+    isSubmitting?: boolean;
+}
+
+export interface PasswordRecoveryFormUpdatableConfigsInterface {
+    /**
+     * Expiry time for the recovery link.
+     */
+    "Recovery.ExpiryTime": any;
```

Comment:
Let's avoid using the `any` datatype

Reply:
addressed with https://github.com/wso2/identity-apps/pull/6418/commits/34b1b04cadee820b1b64ff882b6c969a25b8d4df